frontend/rpm/template.go

```diff
@@ -252,6 +277,10 @@ func (w *specWrapper) BuildSteps() fmt.Stringer {
 
 	fmt.Fprintln(b, "set -e")
 
+	if w.Spec.HasGomods() {
```

Comment:
I'm not sure if this needs to be addressed in this PR since there is only one generator right now, but maybe if we add more generators in the future, it would be worth thinking about an interface for the common actions all generators have to take.  I.e., we could add hooks to allow generators to register handlers for verifying the spec to make sure the generator can be applied, generating extra LLB for the worker state, modifying the RPM spec as it's being built, etc. Just an idea

Reply:
Yep, definitely on my radar.